src/corppa/poetry_detection/annotation/recipe.py

```diff
@@ -0,0 +1,172 @@
+"""
+This module provides custom recipes for Prodigy annotation. They were
+created with page-level text annotation in mind, and support annotating
+text with a reference image displayed beside the text (`annotate_page_text`),
+or annotating both text and image side by side (`annotate_text_and_image`).
+
+Referenced images must be served out independently for display.
+
+Example use:
+```
+prodigy annotate_page_text poetry_spans poetry_pages.jsonl --label POETRY,PROSODY -F ../corppa/poetry_detection/annotation/recipe.py --image-prefix http://localhost:8000/
+prodigy annotate_text_and_image poetry_text_image poetry_pages.jsonl --label POETRY,PROSODY -F ../corppa/poetry_detection/annotation/recipe.py --image-prefix http://localhost:8000/
+"""
+
+from prodigy.core import Arg, recipe
+from prodigy.components.loaders import JSONL
+import spacy
+
+from pathlib import Path
+
+#: reference to current directory, for use as Prodigy CSS directory
+CURRENT_DIR = Path(__file__).parent.absolute()
+
+#: common prodigy configurations for both recipes; copy and add blocks and labels
+PRODIGY_COMMON_CONFIG = {
+    "show_flag": True,  # show flag button to mark weird/difficult examples
+    "hide_newlines": False,  # ensure newlines are shown \n
+    "allow_newline_highlight": True,  # allow highlighting of newlines \n
+    "honor_token_whitespace": True,  # reflect whitespace accurately (e.g. in case of leading/trailing spaces)
+    "custom_theme": {
+        "labels": {
+            # trying to use options from PPA webapp color scheme,
+            # but may not be so great in Prodigy UI.
+            # azure #0788fc seafoam blue #57c4c4 wisteria #9c93c0 pig pink #ed949c
+            "POETRY": "#57c4c4",  # label color for POETRY
+        },
+        "hide_true_newline_tokens": False,
+    },
+    "global_css_dir": CURRENT_DIR,
+}
```

Comment:
Some additional settings we may want:
- `"show_whitespace": true`
- `"instructions": [text file, perhaps excerpt from handbook?]`
- `"allow_work_stealing": false` to ensure that annotators can't "steal" each others loads (hopefully not an issue when they both need to annotate everything)

For annotation assignments we should set either (I believe these settings are meant to be used with sessions):
1. `"feed_overlap": true` so that every session should annotate *all* pages*, or
2. `"annotations_per_task": 2` so that every page gets at least 2 annotations (although I'm not sure allocation works in this case)
I think we want (1) at the moment, but (2) could become useful if we have more annotators (and don't want/need them to annotate the same things)

Reply:
On batch & history sizes, by default it's 10. I think this is just what the web app will serve at one time.

  `"batch_size": 10,
  "history_size": 10,`